src/Uno.UI.Runtime.Skia.Wpf/Rendering/SoftwareWpfRenderer.cs

```diff
@@ -50,6 +50,11 @@ public void Render(DrawingContext drawingContext)
 
 		using var _ = _fpsHelper.BeginFrame();
 
+		while (_host.RootElement is { } rootElement && (rootElement.IsArrangeDirtyOrArrangeDirtyPath || rootElement.IsMeasureDirtyOrMeasureDirtyPath))
+		{
+			rootElement.UpdateLayout();
+		}
```

Comment:
This really feels like it could be placed in a central location. Could `RenderRootVisualAndReturnPath` be a candidate?

Reply:
It really does, but all the composition stuff have no visibility into Uno.UI so we're forced to do it this way.